File: algorithms/1909 - Remove One Element to Make the Array Strictly Increasing.py
class Solution:
    def canBeIncreasing(self, nums: List[int]) -> bool:
        for i in range(len(nums)):
            if self.isIncreasing([nums[j] for j in range(len(nums)) if j != i]):
                return True
        return False
    
    def isIncreasing(self, nums: List[int]) -> bool:
        for i in range(len(nums)):
            if i > 0 and nums[i] <= nums[i-1]:
                return False
        return True

File: algorithms/1909 - Remove One Element to Make the Array Strictly Increasing.ts
function canBeIncreasing(nums: number[]): boolean {
    for (let i = 0; i < nums.length; i++)
        if (isIncreasing(nums.filter((n, j) => j !== i))) return true;
    return false;
};

const isIncreasing = (nums: number[]) => {
    for (let i = 0; i < nums.length; i++)
        if (nums[i] <= nums[i - 1]) return false;
    return true;
}
